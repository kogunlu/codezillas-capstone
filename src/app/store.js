import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';
import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
    user: userReducer,
  },
});
