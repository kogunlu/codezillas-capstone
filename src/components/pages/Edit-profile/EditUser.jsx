import React from 'react';
import userAvatarBlack from '../Signup/Signup_picture/user.jpg';
import UserForm from './forms/UserForm';
import editUserAvatar from './pictures/edit-user.jpg';

function EditUser() {
  return (
    <div className="flex flex-col justify-start items-center h-full my-5 gap-5">
      <div className="w-5/6">
        <p className="text-center text-red-500 font-semibold text-lg">
          Please fill al the fields with correct and valid details to complete
          your profile.
        </p>
      </div>
      <div className="w-5/6 h-full flex flex-col justify-center items-center gap-5 lg:flex-row lg:justify-between lg:items-start lg:gap-0">
        <div className="w-2/6 h-full">
          <span className="flex justify-center ">
            <img
              src={userAvatarBlack}
              alt="userAvatar"
              className="h-60 xl:h-80 -z-10 "
            />
          </span>
          <button
            type="button"
            className="z-20 -mt-10 w-full flex justify-center"
          >
            <img
              src={editUserAvatar}
              alt="editUser"
              className="h-12 border-black border-2 rounded-full"
            />
          </button>
        </div>

        <div className="w-full lg:w-3/6 ">
          <UserForm />
        </div>
      </div>
    </div>
  );
}

export default EditUser;
