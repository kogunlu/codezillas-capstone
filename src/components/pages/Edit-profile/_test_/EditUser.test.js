import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../app/store';
import EditUser from '../EditUser';

it('Renders correctly for EditUser Component', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <EditUser />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
