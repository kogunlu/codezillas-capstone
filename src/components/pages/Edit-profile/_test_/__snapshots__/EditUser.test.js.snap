// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly for EditUser Component 1`] = `
<div
  className="flex flex-col justify-start items-center h-full my-5 gap-5"
>
  <div
    className="w-5/6"
  >
    <p
      className="text-center text-red-500 font-semibold text-lg"
    >
      Please fill al the fields with correct and valid details to complete your profile.
    </p>
  </div>
  <div
    className="w-5/6 h-full flex flex-col justify-center items-center gap-5 lg:flex-row lg:justify-between lg:items-start lg:gap-0"
  >
    <div
      className="w-2/6 h-full"
    >
      <span
        className="flex justify-center "
      >
        <img
          alt="userAvatar"
          className="h-60 xl:h-80 -z-10 "
          src="user.jpg"
        />
      </span>
      <button
        className="z-20 -mt-10 w-full flex justify-center"
        type="button"
      >
        <img
          alt="editUser"
          className="h-12 border-black border-2 rounded-full"
          src="edit-user.jpg"
        />
      </button>
    </div>
    <div
      className="w-full lg:w-3/6 "
    >
      <div>
        <p
          className="text-4xl font-bold px-5 text-center lg:text-start"
        >
          PROFILE INFO
        </p>
        <form
          className="w-full h-full py-10 flex flex-col justify-between gap-5 items-center px-2 md:px-5"
          onSubmit={[Function]}
        >
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Full Name
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="fullName"
                name="fullName"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Education Level
              </p>
            </span>
            <div
              className="w-8/12"
            >
              <select
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base italic text-gray-400"
                onBlur={[Function]}
                onChange={[Function]}
                value="none"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Please select your education level
                </option>
                <option
                  value="primary"
                >
                  Primary School (5 years)
                </option>
                <option
                  value="secondary"
                >
                  Secondary School (8 years)
                </option>
                <option
                  value="highschool"
                >
                  Highschool (12 years)
                </option>
                <option
                  value="university"
                >
                  University Degree (16 years)
                </option>
                <option
                  value="master"
                >
                  Masters Degree or above (20+ years)
                </option>
              </select>
              <div
                className="text-red-400 italic text-sm lg:text-base"
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Hobbies
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="hobbies"
                name="hobbies"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Family Size
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <div
                className="w-full flex justify-start gap-5"
              >
                <input
                  className="border shadow-md w-2/12 h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-center"
                  id="family"
                  name="family"
                  onChange={[Function]}
                  type="number"
                  value="0"
                />
                <span
                  className="w-5/12 flex items-center"
                >
                  <p>
                    Member(s)
                  </p>
                </span>
              </div>
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Gender
              </p>
            </span>
            <div
              className="w-8/12"
            >
              <select
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base italic text-gray-400"
                onBlur={[Function]}
                onChange={[Function]}
                value="none"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Please select your gender
                </option>
                <option
                  value="male"
                >
                  Male
                </option>
                <option
                  value="female"
                >
                  Female
                </option>
                <option
                  value="transsexual"
                >
                  Transsexual
                </option>
                <option
                  value="noSpecify"
                >
                  I don\`t want to specify
                </option>
              </select>
              <div
                className="text-red-400 italic text-sm lg:text-base"
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Birth Date
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-white focus:text-black"
                id="birthdate"
                max="2022-12-31"
                min="1900-01-01"
                name="birthdate"
                onBlur={[Function]}
                onChange={[Function]}
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Email
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Phone
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="phone"
                maxLength="16"
                name="phone"
                onBlur={[Function]}
                onChange={[Function]}
                pattern="[0-9]{2} [0-9]{3} [0-9]{3} [0-9]{2} [0-9]{2}"
                placeholder="90 555 999 99 99"
                type="tel"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Upload ID
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col relative"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="id"
                name="id"
                onChange={[Function]}
                type="number"
                value=""
              />
              <svg
                className="absolute right-5 top-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
                />
              </svg>
            </div>
          </div>
          <div
            className="w-full my-5"
          >
            <p
              className="text-4xl font-bold text-left"
            >
              Security
            </p>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Password
              </p>
            </span>
            <div
              className="w-8/12 h-20 flex flex-col relative"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
              <svg
                className="absolute right-5 top-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19,23 L19,11 L5,11 L5,23 L19,23 Z M12,15 L12,19 M17,11 L17,7 C17,4 17,1 12,1 C7,1 7,4 7,7 L7,11"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Confirm Password
              </p>
            </span>
            <div
              className="w-8/12 h-20 flex flex-col relative"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="confirmPassword"
                name="confirmPassword"
                onChange={[Function]}
                type="password"
                value=""
              />
              <svg
                className="absolute right-5 top-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19,23 L19,11 L5,11 L5,23 L19,23 Z M12,15 L12,19 M17,11 L17,7 C17,4 17,1 12,1 C7,1 7,4 7,7 L7,11"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </div>
          </div>
          <div
            className="w-full flex flex-col flex-wrap items-center gap-5 md:flex-row md:justify-between"
          >
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              type="submit"
              value="SAVE CHANGES"
            />
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-red-600 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              onClick={[Function]}
              type="button"
              value="DELETE ACCOUNT"
            />
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              onClick={[Function]}
              type="button"
              value="CANCEL"
            />
          </div>
          <div
            className="w-full my-5"
          >
            <p
              className="text-4xl font-bold text-left"
            >
              Payment Methods & Tickets
            </p>
          </div>
          <div
            className="w-full flex flex-colitems-center md:flex-row md:justify-start gap-2"
          >
            <div
              className="w-full md:w-8/12 flex flex-col items-center md:items-start justify-between gap-3"
            >
              <p
                className="w-10/12 md:w-5/12 text-center md:text-start"
              >
                3 Cards Added
              </p>
              <input
                className="py-2 px-2 w-10/12 md:w-8/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
                onClick={[Function]}
                type="button"
                value="SHOW CARDS"
              />
            </div>
            <div
              className="w-full md:w-8/12 flex flex-col items-center md:items-start justify-between gap-3"
            >
              <p
                className="w-10/12 md:w-8/12 text-center md:text-start"
              >
                10 Tickets Remaining
              </p>
              <input
                className="py-2 px-2 w-10/12 md:w-8/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
                onClick={[Function]}
                type="button"
                value="BUY TICKETS"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
